Add RightSideComparisonResult as factResult

thorin8k opened the pull request:
Hello!

I noticed the issue mentioned in #310, which is important for my use case, so I tried to address it with this pull request.

My implementation approach is straightforward. When setting the factResult for a condition, if the value property is an object, I include the rightSideValue as the factResult.

I'm eager to see this merged, as it is of great interest to me.

chris-pardy commented:
Conceptually this is fine a few things

It should be in a field like valueResult not overriding the factResult 

You should set the field unconditionally

Make sure you update the types and please add some more unit tests

thorin8k commented:
I've followed your advice and added the field unconditionally. 

Also i've updated all acceptance tests to check for the new value and updated the typings.

I've also added an specific test for this feature checking if the valueResult matches.

thorin8k commented:
I've found that the toJSON Serialization was missing and added it. I've tested in a working application and now it correctly returns the value even when serializing to JSON.

chris-pardy reviewed (approved):
@thorin8k Looks great, thanks for the contribution!
